index.js

```diff
@@ -209,9 +222,38 @@
           }
           return isPropertyDefined('name', config) ? (config.name + '.' + commandName) : commandName;
         };
-
+        
+        // Try to read route configuration and log warning if not possible
+        var routes = {};
+        if (readFromRoute) {
+          if (!$injector.has('$route')) {
+            $log.warn('$route service is not available. Make sure you have included ng-route in your application dependencies.');
+          } else {
+            var $route = $injector.get('$route');
+            routes = $route.routes;
+          }
+        }
         var getUrl = function () {
-          var url = trackUrlParams ? $location.url() : $location.path();
+          // Using ngRoute provided tracking urls
+          var url = $location.url();
+          var trackUrl;
+          Object.keys(routes).forEach(function (key) {
```

Comment:
Instead of checking all of the route objects and storing the routes variable, you can use the current route property:

```javascript
if (readFromRoute) {
  if (('regexp' in $route.current) && $route.current.regexp.test($location.url())) {
    if ('pageTrack' in $route.current) {
      return route.pageTrack;
    }
  }
}
```